utilities/test_suite/HOST/Tensor_host_audio.cpp

```diff
@@ -0,0 +1,206 @@
+    double maxWallTime = 0, minWallTime = 500, avgWallTime = 0;
+    string testCaseName;
+    printf("\nRunning %s %d times (each time with a batch size of %d images) and computing mean statistics...", func.c_str(), numRuns, batchSize);
+    for (int perfRunCount = 0; perfRunCount < numRuns; perfRunCount++)
+    {
+        for (int iterCount = 0; iterCount < noOfIterations; iterCount++)
+        {
+            // Read and decode audio and fill the audio dim values
+            if (inputBitDepth == 2)
+                read_audio_batch_and_fill_dims(srcDescPtr, inputf32, audioFilesPath, iterCount, srcLengthTensor, channelsTensor);
+
+            double startWallTime, endWallTime;
+            double wallTime;
+            switch (testCase)
+            {
+                case 0:
+                {
+                    testCaseName = "non_silent_region_detection";
+                    Rpp32f detectedIndex[batchSize];
+                    Rpp32f detectionLength[batchSize];
+                    Rpp32f cutOffDB = -60.0;
+                    Rpp32s windowLength = 2048;
+                    Rpp32f referencePower = 0.0f;
+                    Rpp32s resetInterval = 8192;
+
+                    startWallTime = omp_get_wtime();
+                    if (inputBitDepth == 2)
+                        rppt_non_silent_region_detection_host(inputf32, srcDescPtr, srcLengthTensor, detectedIndex, detectionLength, cutOffDB, windowLength, referencePower, resetInterval, handle);
+                    else
+                        missingFuncFlag = 1;
+
+                    // QA mode - verify outputs with golden outputs. Below code doesn’t run for performance tests
+                    if (testType == 0)
+                        verify_non_silent_region_detection(detectedIndex, detectionLength, testCaseName, batchSize, audioNames, dst);
+
+                    break;
+                }
+                default:
+                {
+                    missingFuncFlag = 1;
+                    break;
+                }
+            }
+
+            endWallTime = omp_get_wtime();
+            if (missingFuncFlag == 1)
+            {
+                printf("\nThe functionality %s doesn't yet exist in RPP\n", func.c_str());
+                return -1;
+            }
+
+            wallTime = endWallTime - startWallTime;
+            maxWallTime = std::max(maxWallTime, wallTime);
+            minWallTime = std::min(minWallTime, wallTime);
+            avgWallTime += wallTime;
+        }
+    }
+    rppDestroyHost(handle);
+
+    // performance test mode
+    if (testType == 1)
+    {
+        // Display measured times
+        maxWallTime *= 1000;
+        minWallTime *= 1000;
+        avgWallTime *= 1000;
+        avgWallTime /= (numRuns * noOfIterations);
+        cout << fixed << "\nmax,min,avg wall times in ms/batch = " << maxWallTime << "," << minWallTime << "," << avgWallTime;
+    }
+
+    cout<<endl;
+
+    // Free memory
+    free(srcLengthTensor);
```

Comment:
this file is very unorganized code. We need to have better code for testing which is more readable. For test_case, it is better for the user to give a string as an option instead of a number

Reply:
@rrawther I have added name-based testing via Ctest command in this PR, will talk about it.

But the CPP file here is primarily run by the user using the Python script and not directly. 
Basically, the Python script is user-facing, so we could possibly have a --case_name_list option in the py?
(where user can pass say --case_name_list non_silent_region spectrogram to run the desired tests, but the string should be an exact match) @sampath1117